ocfweb/stats/semester_job.py

```diff
@@ -0,0 +1,109 @@
+from datetime import date
+from datetime import datetime
+from datetime import timedelta
+
+import ocflib.printing.quota as quota
+from django.http import HttpResponse
+from matplotlib.figure import Figure
+from matplotlib.ticker import MaxNLocator
+from ocflib.printing.quota import WEEKDAY_QUOTA
+from ocflib.printing.quota import WEEKEND_QUOTA
+
+from ocfweb.component.graph import canonical_graph
+from ocfweb.component.graph import plot_to_image_bytes
+
+ONE_DAY = timedelta(days=1)
+MONTH_INDEX = 1
+SEMESTER_CUTOFF_MONTH = 8
+
+year = datetime.today().year
+spring_start = date(year, 1, 1)
+fall_start = date(year, 8, 1)
+
+
+def get_start_end():
+    """reutnr a tuple of startdate and endate of the semester based on today"""
+    day = datetime.today().timetuple()
+    if day[MONTH_INDEX] < SEMESTER_CUTOFF_MONTH:
+        return (spring_start, fall_start - ONE_DAY)
+    else:
+        return (fall_start, spring_start - ONE_DAY)
```

Comment:
The problem of putting start_date and end_date in constants is that we have to update the years. So I think the utility function that gives the interval is sufficient.